### PR TITLE
textures: jsdoc comments & documentation

### PR DESCRIPTION
**Pull request by belom88**

### CODE REVIEW

File: modules/textures/src/lib/parsers/crunch-module-loader.js [hunk 1]
```diff
@@ -11,6 +11,11 @@ export async function loadCrunchModule(options) {
   return loadCrunch(options);
 }
 
+/**
+ * Load crunch decoder module
+ * @param {any} options - Loader options
+ * @returns {Promise<any>} Promise of Module object
```
Comment (ibgreen) [inline modules/textures/src/lib/parsers/crunch-module-loader.js:17]: Nit: Better type than any?

File: modules/textures/src/lib/parsers/parse-compressed-texture.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { TextureLevel } from "@loaders.gl/textures/types/textrure";
```
Comment (ibgreen) [inline modules/textures/src/lib/parsers/parse-compressed-texture.d.ts:1]:
```suggestion
import { TextureLevel } from "@loaders.gl/textures/types/texture";
```

Typo `textrure` -> `texture` please search replace.

File: modules/textures/src/lib/parsers/parse-dds.d.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+export function parseDDS(data: ArrayBuffer): TextureLevel[];
+
+/**
+ * DXT1 applicable function to calculate level size
+ * @param width - level width
+ * @param height - level height
+ * @returns level size in bytes
+ */
+export function getDxt1LevelSize(width: number, height: number): number;
```
Comment (ibgreen) [inline modules/textures/src/lib/parsers/parse-dds.d.ts:23]: Why do the last two need to be exported? For testing?  Maybe add a separator stating that?

File: modules/textures/src/lib/parsers/parse-dds.js [hunk 4]
```diff
@@ -64,19 +64,24 @@ export function parseDDS(data) {
...
 export function getDxtXLevelSize(width, height) {
   return ((width + 3) >> 2) * ((height + 3) >> 2) * 16;
 }
+
+/**
+ * Convert every byte of Int32 value to char
+ * @param {number} value - Int32 number
+ * @returns {string} string of 4 characters
+ */
+function _int32ToFourCC(value) {
```
Comment (ibgreen) [inline modules/textures/src/lib/parsers/parse-dds.js:80]: We normally don't use underscore on private functions as the lack of an export keyword is enough.

File: modules/textures/src/lib/utils/extract-mipmap-images.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-export function sliceLevels(data, options) {
+export function extractMipmapImages(data, options) {
```
Comment (ibgreen) [inline modules/textures/src/lib/utils/extract-mipmap-images.js:1]: Nit: Sadly `.d.ts` files only checks calls, not definitions. There is a way to import the types so we get the definitions checked against the types, see

https://github.com/visgl/loaders.gl/blob/master/modules/loader-utils/src/lib/binary-utils/buffer-utils.js#L1-L5

**Review by ibgreen — APPROVED** [review]
I like the push to use internal types here. Great start!

File: modules/textures/src/lib/parsers/crunch-module-loader.js (hunk 1 above)
Comment (belom88) [reply modules/textures/src/lib/parsers/crunch-module-loader.js:17]: It is too complicated. The module is pretty large object.

File: modules/textures/src/lib/parsers/parse-dds.d.ts (hunk 3 above)
Comment (belom88) [reply modules/textures/src/lib/parsers/parse-dds.d.ts:23]: These functions  are used in `parse-crunch.js`. Crunch files are "DXT" after decompression and we need for these functions to extract levels.

**Pull request merged**